### ISSUE TITLE
"Edit on github" expects "docs" for content location and ignores "docs_dir" value

### ISSUE THREAD

**Issue by fedorov** [mkdocs/mkdocs#1675]
I had content in a directory called different from "docs". I had it properly set in the docs_dir in mkdocs.yml. However, the "edit on github" button still directed me to an (invalid) location that assumed "docs" is the home of my content.

Pointer to the repository and commit renaming content directory to "docs" to work around this bug: https://github.com/QIICR/DICOM4QI/commit/a1160c4a21c70d42de2939da1520f52aa0c9b736

I am using "material" theme, not sure if this is theme-specific.

**Comment by waylan** [comment]
This is [hardcoded as the default][1] for the [edit_uri][2] setting. While it might be unsurprising if part of the path was derived from the `docs_dir` setting, the documentation notes that the default value assumes the normal defaults for GitHub. If you are using non-standard stuff, then you must manually set the `edit_uri`. The docs specifically call out using a non-standard branch, but a different directory could fit just as well.

Also, note that the `docs_dir` is relative to the config file, while the value of `edit_uri` needs to be relative to the root of the GitHub project. If the config file is not at the root of the project, then the value of `docs_dir` would not work here.

In other words, I'm not inclined to change this. The existing setting is easy enough to change for those who need it and the hardcoded default is clearly documented.

[1]: https://github.com/mkdocs/mkdocs/blob/master/mkdocs/config/config_options.py#L302
[2]: https://github.com/mkdocs/mkdocs/blob/master/mkdocs/config/config_options.py#L302

**Comment by fedorov** [comment]
Thanks for the explanation! Now that I know it, I would know how to deal with this. But I missed this on my read of the documentation. Maybe it can be addressed by mentioning this setting somewhere more prominent. Or hopefully other users are more attentive than me...

**Comment by waylan** [comment]
Well, `edit_uri` is *the setting* which controls the 'Edit on GitHub' link and there is a **Note** in a big blue box calling out the behavior. Not sure how much more prominent we can make it. 

That said, it is not uncommon for users to miss the relevant setting for the problem they are having. My assumption is that users don't actually read through the entire list of configuration settings and don't realize a setting already exists to address their need. Sure, we could highlight a few common ones elsewhere, but I have the sense that they all get missed about equally. And we can't highlight them all effectively.

**Comment by fedorov** [comment]
> Not sure how much more prominent we can make it.

@waylan here are few suggestions:
 - mention it in the front page documentation in https://www.mkdocs.org/
 - populate it to the default value in the `mkdocs.yml` generated by `mkdocs new` (as aside, I was quite surprised to see the auto-generated content to just contain `site_name` - quite a difference as compared to the boilerplate provided by sphinx)

Speaking for myself, it was just non-intuitive that I need to do something extra to allow editing of the content page that (I thought) is well-defined by its URL. I never even thought to look for that setting. If I had it in the boilerplate yml, I think it would raise the flag for me.

**Comment by fedorov** [comment]
@waylan - just 2c from a (confused or attention-lacking) user. 

I will not be offended if you close this issue.
